test/test.hpp

```diff
@@ -251,13 +254,14 @@ namespace tts
   template<eve::simd_value W> auto poison(W data)
   {
     using v_t = eve::element_type_t<W>;
+    using c_t  = eve::cardinal_t<W>;
 
     // Add garbage at the end of sub-native registers
     // For emulated type, there is no such gap so we don't do anything
     if constexpr( (W::size() < eve::fundamental_cardinal_v<v_t>) && !eve::has_emulated_abi_v<W> )
     {
       using p_t   = eve::as_arithmetic_t<eve::as_integer_t<v_t, unsigned>>;
-      using ftype = eve::as_wide_t<v_t, eve::fundamental_cardinal_t<v_t>>;
+      using ftype = eve::as_wide_t<v_t, c_t>;
```

Comment:
Wait - this can't be right. And why this change, fundamental is 1 and size is never less

Reply:
Changed to separate case for RISC-V RVV. The problem for rvv is that for type with fundamental cardinal we use one underlying type (with `m1` at the end), and for smaller types we can use other underlying type (to use part of real vector register, if possible).